From 5a7c3999190d640f43d9d1f6430c723fab8ebe15 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sat, 9 Jan 2021 01:44:55 -0500
Subject: [PATCH] [mobile-nixos]: exynos/decon_7880: Force unblank on probe

This removes the requirement to blank/unblank on boot, and makes the
kernel console available ASAP.

An issue is present, but was also present with the previous workaround,
is that there is a big red flash of stuff on display. Though it is
quickly removed on redraw.
---
 drivers/video/fbdev/exynos/decon_7880/decon_core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/fbdev/exynos/decon_7880/decon_core.c b/drivers/video/fbdev/exynos/decon_7880/decon_core.c
index 6bf11d5245b..543efe8c41c 100755
--- a/drivers/video/fbdev/exynos/decon_7880/decon_core.c
+++ b/drivers/video/fbdev/exynos/decon_7880/decon_core.c
@@ -4453,6 +4453,10 @@ decon_init_done:
 	decon->log_cnt = 0;
 	decon_info("decon%d registered successfully\n", decon->id);
 
+	decon_info("decon%d forcing unblank after successful probe...\n", decon->id);
+	decon_blank(FB_BLANK_UNBLANK, fbinfo);
+	decon_info("decon%d ... done with forcing unblank.\n", decon->id);
+
 	return 0;
 
 fail_update_thread:
-- 
2.29.2

